tls_heartbeat.rflx

```diff
@@ -2,16 +2,18 @@ package TLS_Heartbeat is
 
    type Message_Type is (HEARTBEAT_REQUEST => 1, HEARTBEAT_RESPONSE => 2) with Size => 8;
    type Length is range 0 .. 2**14 - 20 with Size => 16;
+   type TLS_Plaintext_Length is range 3 .. 2 ** 14 with Size => 16;
 
-   type Heartbeat_Message is
+   type Heartbeat_Message (TLS_Plaintext_Length : TLS_Plaintext_Length) is
       message
          Message_Type : Message_Type;
          Payload_Length : Length;
          Payload : Opaque
             with Size => Payload_Length * 8;
          Padding : Opaque
-            with Size => Message'Size - Payload_Length * 8 - 24
-            if Message'Size / 8 <= 2**14 and Padding'Size / 8 >= 16;
+            with Size => (TLS_Plaintext_Length - Payload_Length - (Message_Type'Size + Payload_Length'Size) / 8) * 8
+            if TLS_Plaintext_Length >= Payload_Length + (Message_Type'Size + Payload_Length'Size) / 8
+               and Padding'Size >= 16;
```

Comment:
- The check of the padding size is wrong, should be: `Padding'Size / 8 >= 16`.
- I think the size aspect would be easier like this: `(TLS_Plaintext_Length - Payload_Length) * 8 - Message_Type'Size - Payload_Length'Size`.